File: apps/qizhi/mobile/MentorAI-Android/app/src/main/kotlin/com/mentorai/app/user/UserProfile.kt
package com.mentorai.app.user

import com.mentorai.app.support.ServerDate
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Mirrors the iOS `UserProfile`: every field is optional because the backend may omit them.
 * Display fallbacks live in `displayName` / `formattedCreateTime`.
 */
@Serializable
data class UserProfile(
    val id: String? = null,
    @SerialName("zju_id") val zjuId: String? = null,
    val name: String? = null,
    val department: String? = null,
    val phone: String? = null,
    val email: String? = null,
    @SerialName("create_time") val createTime: String? = null,
) {
    val displayName: String
        get() = listOf(name, zjuId, id).firstOrNull { !it.isNullOrBlank() } ?: "—"

    val formattedCreateTime: String?
        get() = createTime?.takeIf { it.isNotBlank() }?.let { ServerDate.absolute(it) }
}

File: apps/qizhi/mobile/MentorAI-iOS/MentorAI/User/UserProfile.swift
import Foundation

struct UserProfile: Codable, Equatable {
    var id: String?
    var zjuID: String?
    var name: String?
    var department: String?
    var phone: String?
    var email: String?
    var createTime: String?

    enum CodingKeys: String, CodingKey {
        case id
        case zjuID = "zju_id"
        case name
        case department
        case phone
        case email
        case createTime = "create_time"
    }
}

extension UserProfile {
    var displayName: String {
        if let name, !name.isEmpty { return name }
        if let zjuID, !zjuID.isEmpty { return zjuID }
        if let id, !id.isEmpty { return id }
        return "—"
    }

    var formattedCreateTime: String? {
        guard let raw = createTime, !raw.isEmpty else { return nil }
        return ServerDate.absolute(raw)
    }
}
